packages/visual-editor/src/internal/puck/components/ThemeHeader.tsx

```diff
@@ -177,7 +179,7 @@ export const ThemeHeader = (props: ThemeHeaderProps) => {
               await onPublishTheme();
             }}
           >
-            Publish
+            {`Update ${totalPageCount} ${totalPageCount === 1 ? "Page" : "Pages"}`}
```

Comment:
Similar comment as the CR - both numbers should derive from the same counts (entity), not one is for entityCount and one is for pageCount